spec/services/curator/filestreams/derivatives_service_spec.rb

```diff
@@ -13,36 +13,74 @@
   end
 
   describe '#call' do
-    subject!(:file_set) do
-      fs = nil
-      json_data = file_set_json.dup
-      json_data['files'] = files_json
-      json_data['file_set_of']['ark_id'] = parent_obj.ark_id
-      json_data['exemplary_image_of'] = []
-      json_data['exemplary_image_of'][0] = { 'ark_id' => parent_obj.ark_id }
-      json_data['exemplary_image_of'][1] = { 'ark_id' => parent_col.ark_id }
-      VCR.use_cassette('services/filestreams/file_set_for_derivatives') do
-        _, fs = handle_factory_result(Curator::Filestreams::FileSetFactoryService, json_data)
+    let!(:parent_col) { create(:curator_collection) }
+    let!(:parent_obj) { create(:curator_digital_object, admin_set: parent_col) }
+
+    context 'image_file_set' do
+      subject!(:image_file_set) do
+        fs = nil
+        json_data = image_file_set_json.dup
+        json_data['files'] = image_files_json
+        json_data['file_set_of']['ark_id'] = parent_obj.ark_id
+        json_data['exemplary_image_of'] = []
+        json_data['exemplary_image_of'][0] = { 'ark_id' => parent_obj.ark_id }
+        json_data['exemplary_image_of'][1] = { 'ark_id' => parent_col.ark_id }
+        VCR.use_cassette('services/filestreams/image_file_set_for_derivatives') do
+          _, fs = handle_factory_result(Curator::Filestreams::FileSetFactoryService, json_data)
+        end
+        fs
       end
-      fs
-    end
 
-    subject!(:called_result) do
-      res = nil
-      VCR.use_cassette('services/filestreams/derivatives') do
-        res = described_class.call(file_set.avi_file_class, avi_payload: file_set.avi_payload)
+      subject!(:image_called_result) do
+        res = nil
+        VCR.use_cassette('services/filestreams/image_derivatives') do
+          res = described_class.call(image_file_set.avi_file_class, avi_payload: image_file_set.avi_payload)
+        end
+        res
+      end
+
+      let!(:image_file_set_json) { load_json_fixture('image_file_set_2', 'file_set') }
+      let!(:image_files_json) { load_json_fixture('image_file_2', 'files') }
+
+      it 'expects the result to be successful' do
+        expect(image_called_result).to be_truthy
+        expect(image_called_result).to be_a_kind_of(Hash).and have_key(image_file_set.avi_file_class)
       end
-      res
     end
 
-    let!(:parent_col) { create(:curator_collection) }
-    let!(:parent_obj) { create(:curator_digital_object, admin_set: parent_col) }
-    let!(:file_set_json) { load_json_fixture('image_file_set_2', 'file_set') }
-    let!(:files_json) { load_json_fixture('image_file_2', 'files') }
+    context 'video_file_set' do
+      subject!(:video_file_set) do
+        fs = nil
+        json_data = video_file_set_json.dup
+        json_data['files'] = video_files_json
+        awesome_print video_files_json
```

Comment:
We should remove this.

Also, I find specs a little more understandable when the `let!` declarations are above the expectations, any objection to moving them?